### DISCUSSION TITLE
Can I use this to visualize my own data (other than what it offers)?

### DISCUSSION THREAD

**Discussion by avinal** [arl/statsviz#102] [Q&A] [answered]
I have an application spitting some 4 data points in real-time. Let's say for example the number of requests received by the application and the number of responses. How can I use this library to visualize that? 

If there is an example, that would be very helpful. Thanks.

**Comment by arl** [comment, marked as answer]
Hi @avinal
Unfortunately at the moment Statsviz doesn't expose an API for users to show their data. That's something I considered when building Statsviz so it should not require an exaggerated amount of work, but for now that work hasn't been done. 
I'll update that discussion when I start to work on it.

**Reply by avinal** [reply]
Thanks @arl for the quick response. This feature would be extremely helpful.

P.S. - If you happen to know any go library that does this please tell me. I have been searching for 2 days and have only been able to find some terminal-based go libraries for the purpose. I need a browser one.

**Reply by arl** [reply]
Nothing comes to mind. Of course there's prometheus+grafana which is a very powerful combo but I guess you were looking at something a bit less involved.

**Reply by avinal** [reply]
Yes, I was looking for something much more straightforward. I achieved this by using WebSockets and Google Charting library. You can see the implementation [here](https://github.com/avinal/snowgauge). The code right now is held by tapes and bandages though 😅 , will improve. 

Thanks for all the help.

**Reply by arl** [reply]
Nice.
For the record I've been working since yesterday on the new Statsviz API where I expose functions and methods allowing users to add their own plots. No ETA since I'm working on this on my free time but I'm really close. It's probable this will be available next week. I'll update this thread

**Reply by arl** [reply]
User plots have been available for quite some time in Statsviz and I forgot about this thread.

**Reply by avinal** [reply]
Hey, thanks; even I forgot about it, since it was just a weekend project. Will check it out.
